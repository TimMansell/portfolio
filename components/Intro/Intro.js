// import React from "react";

import useScrollFade from "hooks/useScrollFade";

import Hero from "../Hero";
import LearnMore from "components/LearnMore";
import Hi from "components/Hi";
import Location from "components/Location";
import Name from "components/Name";
import Tagline from "components/Tagline";

import styles from "./Intro.module.scss";

export const Intro = () => {
  const scrollFadeStyle = useScrollFade(1.15);

  return (
    <section id="intro" className={styles.background} role="banner">
      <Hero />

      <div className={styles.intro}>
        <Hi />

        <div style={scrollFadeStyle} data-e2e="fading-content">
          <Location />

          <Name />

          <Tagline />
        </div>
      </div>

      <LearnMore />
    </section>
  );
};

export default Intro;
