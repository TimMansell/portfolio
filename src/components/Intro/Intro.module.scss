.intro {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  position: fixed;
  top: 0;
  width: 100%;
  height: 100%;
  text-align: center;
}

.background {
  composes: primary from 'scss/helpers/backgrounds.scss';
}
