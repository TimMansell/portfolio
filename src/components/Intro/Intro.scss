@import 'variables';
@import 'utilities/breakpoints';

.info {
	display: flex;
    align-items: center;
	justify-content: center;
	flex-direction: column;
    position: fixed;
    top: 0;
    width: 100%;
	height: 100%;
	text-align: center;
	

	&__title {
		color: $secondary-text-colour;
		margin: 15px 0;
		padding: 0 25px;
		font-size: 45px;
		font-weight: 900;
		letter-spacing: 5px;
		line-height: 45px;
		text-transform: uppercase;
		text-shadow: 3px 3px darken($primary-colour, 10%);
		opacity: 0;

		animation: title 3s ease-in-out;
		animation-delay: 3.75s;
		animation-fill-mode: forwards;

		@include media-breakpoint-up(md){
			text-shadow: 2px 2px $tertiary-text-colour;
			font-size: 55px;
			line-height: 55px;
		}
	}

	&__text {
		color: $secondary-text-colour;
		font-weight: 300;
		letter-spacing: 1px;
		line-height: 26px;
		text-transform: uppercase;
		text-shadow: 1px 1px darken($primary-colour, 10%);

		@include media-breakpoint-up(md){
			text-shadow: 2px 2px $tertiary-text-colour;
		}

		&--sub-title {
			font-size: 16px;
			opacity: 0;

			animation: location 3s ease-in-out;
			animation-delay: 4s;
			animation-fill-mode: forwards;

			@include media-breakpoint-up(md){
				font-size: 18px;
			}
		}

		&--blurb {
			font-size: 22px;
			display: block;
		
			@include media-breakpoint-up(md){
				font-size: 26px;
			}
		}
	}
}

@keyframes title {
	from {
		opacity: 0;
	}

	to {
		opacity: 1;
	}
}

@keyframes location {
	from {
		opacity: 0;
		transform: translateY(-10px);
	}

	to {
		opacity: 1;
		transform: translateY(0);
	}
}



// @keyframes title {
// 	from {

// 	}

// 	to {
		
// 	}
// }