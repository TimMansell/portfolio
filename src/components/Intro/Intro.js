import React from 'react';
import { Link } from 'react-scroll';

import Hero from '../Hero';
import ScrollFade from '../ScrollFade';
import ShuffleContent from '../ShuffleContent';

import './Intro.scss';

import { library } from '@fortawesome/fontawesome-svg-core';
import { faChevronDown } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

library.add(faChevronDown);

export const Intro = () => {
	return <section id="intro" className="hero bg--primary">
	<Hero blurFrom={3} blurTo={10} />

	<div className="info">
		<ScrollFade fadeMultiplier={1.15}>
			<h4 className="info__text info__text--sub-title">Melbourne, Australia</h4>
			<h1 className="info__title">Hi! I'm Tim Mansell</h1>
			
			<ShuffleContent>
				<h4 className="info__text info__text--blurb">Front-end Engineer</h4>
				<h4 className="info__text info__text--blurb">Javascript Engineer</h4>
				<h4 className="info__text info__text--blurb">React Engineer</h4>
				<h4 className="info__text info__text--blurb">VueJS Engineer</h4>
			</ShuffleContent>
			
		</ScrollFade>
	</div>

	<div className="learn-more">
		<Link className="learn-more__link" to="profile" smooth={true}>
			<FontAwesomeIcon className="learn-more__scroll" icon={faChevronDown} />
		</Link>
	</div>

</section>;
}

export default Intro;