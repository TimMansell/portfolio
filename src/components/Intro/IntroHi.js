import React from 'react';

import './Intro.scss';

export const IntroHi = () => {
	return <div>
		<h1 className="intro-hi">Hi!</h1>
	</div>;
}

export default IntroHi;