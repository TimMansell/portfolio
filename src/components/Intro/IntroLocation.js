import React from 'react';

import './Intro.scss';

export const IntroLocation = () => {
	return <h4 className="info__text info__text--sub-title">Melbourne, Australia</h4>;
}

export default IntroLocation;