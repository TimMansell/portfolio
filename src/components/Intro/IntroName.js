import React from 'react';

import './Intro.scss';

export const IntroName = () => {
	return <h1 className="info__title">I'm Tim Mansell</h1>;
}

export default IntroName;