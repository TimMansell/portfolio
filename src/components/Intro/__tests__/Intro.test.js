import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Intro from '../Intro';

describe('Intro', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<Intro />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<Intro/>).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
