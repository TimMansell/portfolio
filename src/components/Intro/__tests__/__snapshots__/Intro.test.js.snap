// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro should match snapshot 1`] = `
<section
  className="bg--primary"
  id="intro"
  role="banner"
>
  <div
    className="hero"
  >
    <div
      className="hero__img"
      style={Object {}}
    />
  </div>
  <div
    className="intro"
  >
    <p
      className="hi"
    >
      Hi!
    </p>
    <div
      style={Object {}}
    >
      <div
        className="location"
      >
        Melbourne, Australia
      </div>
      <h1
        className="name"
      >
        I'm Tim Mansell
      </h1>
      <div
        className="tagline"
      >
        <div
          className="tagline__item"
        >
          Front-end Engineer
        </div>
        <div
          className="tagline__item"
        >
          Javascript Engineer
        </div>
        <div
          className="tagline__item"
        >
          React Engineer
        </div>
        <div
          className="tagline__item"
        >
          VueJS Engineer
        </div>
      </div>
    </div>
  </div>
  <div
    className="learn-more"
  >
    <a
      className="learn-more__link"
      data-e2e="learn-more-btn"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-down fa-w-14 icon learn-more__scroll icon--sm"
        data-icon="chevron-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
</section>
`;
